### PR TITLE
UHM-5028 Migrate HIV_INTAKE_FORMS table to OpenMRS

### PR DESCRIPTION
**Pull request by brandones**
Creates generic table stager to move tables from Oracle to MySQL

### CODE REVIEW

File: etl/src/main/java/org/pih/hivmigration/etl/sql/TableStager.groovy [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    String mysqlTableName
+
+    TableStager(String oracleTableName) {
+        this.oracleTableName = oracleTableName
+        this.mysqlTableName = "hivmigration_" + oracleTableName.replace("HIV_", "").toLowerCase()
+    }
+
+    @Override
+    void migrate() {
+        def datatypeMap = [VARCHAR2: "TEXT", NUMBER: "NUMERIC"]
```
Comment (mseaton) [inline etl/src/main/java/org/pih/hivmigration/etl/sql/TableStager.groovy:19]: Why map varchar2 to text?  I would think we could construct an actual varchar column of the appropriate length with a bit more code.  If there is no performance or other negative impact to using text, I guess that's fine, but I would assume there is.

**Review by mseaton — APPROVED** [review]
Looks like a nice and helpful class.  A few additional utilities I can think of:
* The ability to exclude columns that are not relevant or used
* The ability to convert columns named *_p and have values of 't' or 'f' into type boolean 1/0
Or not.  We can use this for the cases where we just want to move an entire table over, and choose to do something custom where we need it.

**Comment by mogoodrich** [comment]
I did a quick glance at this but in the interest of divide and conquer I didn't dive in too deeply because looks like Mike has already done so.  But conceptually this seems like a good model, and will allow us to systematically include things like filtering out test patients/encounters like you did.

**Review by cioan — APPROVED** [review]

File: etl/src/main/java/org/pih/hivmigration/etl/sql/TableStager.groovy (hunk 1 above)
Comment (brandones) [reply etl/src/main/java/org/pih/hivmigration/etl/sql/TableStager.groovy:19]: The only negative impact of using TEXT is that you can't index on it. Is that okay?

Comment (brandones) [reply etl/src/main/java/org/pih/hivmigration/etl/sql/TableStager.groovy:19]: https://stackoverflow.com/a/25301046/1464495

**Pull request merged**